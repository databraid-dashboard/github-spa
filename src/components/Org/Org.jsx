/* eslint-disable import/no-named-as-default, no-shadow */
import React from 'react';
import { Grid, Image, List } from 'semantic-ui-react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { renderRepos } from '../../actions/renderActions';
import './Org.css';

export const Org = ({ avatarUrl, orgName, renderRepos }) => (
  <Grid.Column>
    <Image
      className="hoverable"
      onClick={() => renderRepos(orgName)}
      src={avatarUrl}
      size="small"
    />
    <List.Item
      as="a"
      className="hoverable"
      onClick={() => renderRepos(orgName)}
    >
      {orgName}
    </List.Item>
  </Grid.Column>
);

export const mapStateToProps = (state, { orgId }) => {
  const org = state.orgs.orgsById[orgId];
  const { avatarUrl, orgName } = org;
  return {
    avatarUrl,
    orgName,
  };
};

const mapDispatchToProps = dispatch => bindActionCreators({
  renderRepos,
}, dispatch);

Org.propTypes = {
  avatarUrl: PropTypes.string.isRequired,
  orgName: PropTypes.string.isRequired,
  renderRepos: PropTypes.func.isRequired,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Org);
