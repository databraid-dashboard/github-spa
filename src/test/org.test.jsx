import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { Org, mapStateToProps } from '../components/Org/Org';

const state = {
  orgs: {
    ids: [238923429],
    orgsById: {
      238923429: {
        avatarUrl: 'https://avatars1.githubusercontent.com/u/29614654?v=4',
        orgName: 'ShareCastG52',
        reposUrl: 'https://api.github.com/orgs/ShareCastG52/repos',
      },
    },
  },
  loadingOrganizations: false,
};

describe('Org entry component', () => {
  it('should render a component with props as specified ', () => {
    const component = shallow(
      <Org

        avatarUrl={'https://avatars1.githubusercontent.com/u/29614654?v=4'}
        orgName={'ShareCastG52'}
        reposUrl={'https://api.github.com/orgs/ShareCastG52/repos'}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
  it('map\'s given State To Props ', () => {
    const expected = {
      avatarUrl: 'https://avatars1.githubusercontent.com/u/29614654?v=4',
      orgName: 'ShareCastG52',
    };
    expect(mapStateToProps(state, { orgId: 238923429 })).toEqual(expected);
  });
});
