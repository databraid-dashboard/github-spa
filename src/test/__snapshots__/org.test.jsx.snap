// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Org entry component should render a component with props as specified  1`] = `
<GridColumn>
  <Image
    as="img"
    className="hoverable"
    onClick={[Function]}
    size="small"
    src="https://avatars1.githubusercontent.com/u/29614654?v=4"
    ui={true}
  />
  <ListItem
    as="a"
    className="hoverable"
    onClick={[Function]}
  >
    ShareCastG52
  </ListItem>
</GridColumn>
`;
